sentry/src/main/java/io/sentry/SentryClient.java

```diff
@@ -353,22 +353,29 @@ public void captureSession(final @NotNull Session session, final @Nullable Objec
     SentryId sentryId = transaction.getEventId();
 
     if (transaction instanceof SentryTransaction) {
-      final SentryTransaction sentryTransaction =
-          processTransaction((SentryTransaction) transaction);
-      try {
-        final SentryEnvelope envelope =
-            buildEnvelope(sentryTransaction, filterForTransaction(getAttachmentsFromScope(scope)));
-        if (envelope != null) {
-          transport.send(envelope, hint);
-        } else {
+      final SentryTransaction sentryTransaction = (SentryTransaction) transaction;
+
+      if (!sentryTransaction.isFinished()) {
+        options.getLogger().log(SentryLevel.INFO, "Finishing unfinished transaction %s.", sentryId);
+        sentryTransaction.finish();
+      } else {
```

Comment:
is that right? wouldn't it be better to `finish` AND capture the transaction? or am I missing something?

Reply:
ahhhh `finish` calls `captureTransaction` already, guess that's the trick, right?